nav2_costmap_2d/include/nav2_costmap_2d/obstacle_layer.hpp

```diff
@@ -44,7 +44,10 @@
 
 #include "rclcpp/rclcpp.hpp"
 #include "laser_geometry/laser_geometry.hpp"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreorder"
```

Comment:
Let's submit a PR to geometry2/tf2_ros to address this issue. A simple fix for message_filter.h and we can remove the pragmas.

Reply:
Agreed. I'll do that next week.